File: template.py
import sys
import os

# setting path
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__)))))

# importing
import utils

def puzzle(filename, part2):
    # Zero the Accumulator
    score = 0
    
    # Open File
    with open(filename, 'r') as fp:
        while True:
            # Loop over all lines
            line = fp.readline()
            
            # EOF check
            if line == '':
                break   
    
    # Return Accumulator    
    print(score)
    

        
l = [['1', '2'], ['3', '4']]

g = utils.Grid(l)
print(g)
print(g.get(0, 0))
print(g.get((1, 1)))
    
if __name__ == "__main__":
    # Check number of Arguments, expect 2 (after script itself)
    # Args: input file, part number
    if (len(sys.argv) != 3):
        print(f"Error: Expected 2 arguments, got {len(sys.argv) - 1}")
        
    else:
        puzzle(sys.argv[1], sys.argv[2] == '2')

File: cpp/template.cpp
#include <iostream>
#include <fstream>
#include "utils.hpp"
using namespace std;

void puzzle(char *input, bool part2) {
    ifstream fp(input, ios_base::in);
    string line;

    // Loop over file, line by line
    while (getline (fp, line)) {

    }

}

int main(int argc, char *argv[]) {
    // Expect 2 arguments beyond the caller: input file and part number
    if (argc == 3) {
        puzzle(argv[1], argv[2][0] == '2');
    } else {
        cerr << "Error: Expected 2 arguments, got: " << (argc - 1) << endl;
    }

    return 0;
}
